test/integration/scenarios/01.update/magefile.go

```diff
@@ -39,28 +39,27 @@ func Test() error {
 	fmt.Println("Running ", TEST_NAME)
 
 	defer Cleanup()
+	err := Setup()
+	if err != nil {
+		return err
+	}
 
+	// Wait a few secs for symphony cert to be ready;
+	// otherwise we will see error when creating symphony manifests in the cluster
+	// <Error from server (InternalError): error when creating
+	// "/mnt/vss/_work/1/s/test/integration/scenarios/basic/manifest/target.yaml":
+	// Internal error occurred: failed calling webhook "mtarget.kb.io": failed to
+	// call webhook: Post
+	// "https://symphony-webhook-service.default.svc:443/mutate-symphony-microsoft-com-v1-target?timeout=10s":
+	// x509: certificate signed by unknown authority>
+	time.Sleep(time.Second * 10)
 	for _, namespace := range TestNamespaces {
-		err := Setup()
-		if err != nil {
-			return err
-		}
 
-		// Wait a few secs for symphony cert to be ready;
-		// otherwise we will see error when creating symphony manifests in the cluster
-		// <Error from server (InternalError): error when creating
-		// "/mnt/vss/_work/1/s/test/integration/scenarios/basic/manifest/target.yaml":
-		// Internal error occurred: failed calling webhook "mtarget.kb.io": failed to
-		// call webhook: Post
-		// "https://symphony-webhook-service.default.svc:443/mutate-symphony-microsoft-com-v1-target?timeout=10s":
-		// x509: certificate signed by unknown authority>
-		time.Sleep(time.Second * 10)
```

Comment:
Don't we need to sleep() and cleanup() for each namespace if there are multiple?

Reply:
As the comment said, this sleep is to wait for symphony cert to be ready. So I think it's only needed after symphony setup. Different namespaces are where symphony objects - solution, target, instances are created rather than whether symphony pods are deployed.

Reply:
And we want to test the scenario that symphony objects with the same name exist in different namespace at the same time. setup and cleanup for each namespace won't cover this scenario.